### PR TITLE
Modify and add events for subquery, UI

### PR DESCRIPTION
**Pull request by krkzts**
# What
Modified and added events for subquery, UI.

# Draft to be added events
* PnlRealized event
```solidity
event PnlRealized {
	address trader,
	uint256 amount
}
```

*  Event that fires every time the markPrice changes
```solidity
event MarkPriceChanged {
	address market,
	uint256 timestamp,
	uint256 price
}
```

# Check list
- [ ] Is there any missing information that we want to pass from contract to front end or information that we want to store in the db?

### PR Reminders

Be ware of the followings

- [ ] implement deployment script for your changes in deployment repo
- [ ] add verification in hardhat simulation(000-prepare-simulation-check, 902-newMarket-check or 903-simulation-check) and system test(901-system-test) in deployment repo
- [ ] update change log
 
---

1. **Contract**: make sure the code follows our convention, ex: naming, explicit returns, etc.; if uncertain, discuss with others

2. **Test**: make sure tests can cover most normal and edge cases; if not, open follow-up tickets. Also, look out for failed tests and fix them!

3. **CHANGELOG.md**: update when external interfaces are changed

4. Workflow: 
    - Github: assign the pr to yourself; if pairing, can merge directly; else, assign someone to help review
    - Asana: assign the corresponding ticket to yourself and leave the pr link on it for easier follow-ups
    - Discord: if someone is mentioned in this pr, tag on Discord

### CODE REVIEW

File: contracts/interface/IPerpdexExchange.sol [hunk 1]
```diff
@@ -64,15 +64,42 @@ interface IPerpdexExchange {
...
+        address market,
+        int256 amount,
         uint256 positionSize,
+        uint256 positionNotional,
         uint256 liquidationFee,
-        address liquidator
+        address liquidator,
+        int256 oppositeAmountBound,
+        uint256 deadline,
+        uint24 priceLimitNormalOrderRatio,
```
Comment (richmanbtc) [inline contracts/interface/IPerpdexExchange.sol:76]: I think that config is unnecessary here if you add a config change event separately

File: contracts/interface/IPerpdexExchange.sol [hunk 2]
```diff
@@ -64,15 +64,42 @@ interface IPerpdexExchange {
...
+        uint24 priceLimitLiquidationRatio,
+        uint24 mmRatio,
+        uint24 liquidationRewardRatio,
+        uint8 maxMarketsPerAccount,
+        uint24 protocolFeeRatio,
+        bool isBaseToQuote,
+        bool isExactInput,
+        int256 penalty,
```
Comment (richmanbtc) [inline contracts/interface/IPerpdexExchange.sol:84]: redundant because penalty == intliquidatorReward + insuranceFundReward

File: contracts/interface/IPerpdexExchange.sol [hunk 3]
```diff
@@ -81,7 +108,18 @@ interface IPerpdexExchange {
         int256 exchangedPositionNotional,
         int256 openNotional,
         int256 realizedPnl,
-        uint256 priceAfterX96
+        uint256 priceAfterX96,
+        int256 tradingFee,
```
Comment (richmanbtc) [inline contracts/interface/IPerpdexExchange.sol:112]: Trade fees are difficult to define because they are inseparable from slippage. If you want to calculate, it may be good to record the price before swap so that you can see the difference in the average trade price.

File: contracts/interface/IPerpdexExchange.sol (hunk 1 above)
Comment (krkzts) [reply contracts/interface/IPerpdexExchange.sol:76]: I thought it would be better to also have the value of config when you set up the position when saving the position history, but that may indeed be redundant.
Would the config change event be defined as follows?
```solidity
event ConfigChanged{
	int256 oppositeAmountBound,
        uint256 deadline,
        uint24 priceLimitNormalOrderRatio,
        uint24 priceLimitLiquidationRatio,
        uint24 mmRatio,
	uint24 imRatio
        uint24 liquidationRewardRatio,
        uint24 protocolFeeRatio
}
```
Also, is maxMarketsPerAccount, isBaseToQuote, isExactInput unnecessary?

File: contracts/interface/IPerpdexExchange.sol (hunk 2 above)
Comment (krkzts) [reply contracts/interface/IPerpdexExchange.sol:84]: Thanks.
Should we output intliquidatorReward, insuranceFundReward in event and calculate penalty in subquery?

**Comment by krkzts** [comment]
Thank you for your review.

I have a suggestion for an additional event modification, I think we need quote fee in LiquidityChanged event.

**Comment by richmanbtc** [comment]
> cumDeleveragedBaseSharePerLiquidityX96
I gave this name but it's too long so I want to think of a good name

File: contracts/interface/IPerpdexExchange.sol [hunk 4]
```diff
@@ -81,7 +108,18 @@ interface IPerpdexExchange {
...
+        bool isBaseToQuote,
+        bool isExactInput,
+        uint256 oppositeAmountBound,
+        uint256 deadline,
```
Comment (richmanbtc) [inline contracts/interface/IPerpdexExchange.sol:116]: I think this is not required
